### PR TITLE
BAVL-123 intial attempt at wiring in call to email service for sending a court email. Still more work to be done.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
This is only sending one email at the moment.

It is also recording successful emails sent in the notifications audit trail table.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt [hunk 1]
```diff
@@ -40,17 +40,17 @@ class CourtNewBookingEmail(
   userName: String,
   court: String,
   prison: String,
-  preAppointmentInfo: String = "",
+  preAppointmentInfo: String?,
   mainAppointmentInfo: String,
-  postAppointmentInfo: String = "",
-  comments: String?,
+  postAppointmentInfo: String?,
+  comments: String? = "None entered",
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt:46]: Default values for comments and pre/post appointments take from existing BVLS.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/config/EmailConfiguration.kt [hunk 2]
```diff
@@ -54,7 +55,7 @@ abstract class Email(
   comments: String? = "",
 ) {
   private val common = mapOf(
-    "date" to date.toString(),
+    "date" to date.toMediumFormatStyle(),
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/config/EmailConfiguration.kt:58]: This is the style used in the templates.  Note we have to do the formatting here.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/Prison.kt [hunk 3]
```diff
@@ -19,7 +19,7 @@ class Prison(
 
   val description: String,
 
-  // Enabled == Courts/probation cannot self-serve, but we will accept bookings from the prison (via DPS and A&A)
+  // Enabled == false means courts/probation cannot self-serve, but we will accept bookings from the prison (via DPS and A&A)
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/Prison.kt:22]: Thanks

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt [hunk 4]
```diff
@@ -44,4 +77,14 @@ class BookingFacade(
     // Agreed with Tim will be using a ContactService to pull back the necessary contact information related to the booking just created.
     // Multiple contacts will result in multiple emails i.e. one email per contact.
   }
+
+  private fun Collection<BookingContact>.allContactsWithAnEmailAddress() = filter { it.email != null }
+
+  private fun Collection<PrisonAppointment>.courtAppointments() = Triple(pre(), main(), post())
+
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt:84]: This seems a little confusing as they are prison appointments, not court appointments. The only one involving the court (really) is the main hearing. Probably just terminology of calling them courtAppointments?

**Review by timharrison-moj — APPROVED** [review]
LGTM. All the right things happening.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt (hunk 4 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt:96]: Changed.

**Review by timharrison-moj — DISMISSED** [review]
LGTM

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**